src/sentry/tasks/base.py

```diff
@@ -129,6 +129,13 @@ def _wrapped(*args, **kwargs):
 
             return result
 
+        if kwargs.get("soft_time_limit") and not kwargs.get("time_limit"):
+            raise ValueError("time_limit must be set if soft_time_limit is set")
+
+        if kwargs.get("time_limit") is None and kwargs.get("soft_time_limit") is None:
+            kwargs["soft_time_limit"] = int(timedelta(hours=3).total_seconds())
+            kwargs["time_limit"] = int(timedelta(hours=3, seconds=15).total_seconds())
```

Comment:
Should this be a config value instead?

Reply:
got it, i wasn't sure about using those. I'll look into briefly and make sure they work, but can certainly use the config values.